util/topgen.py

```diff
@@ -1196,7 +1196,8 @@ def generate_modules(template_type: str,
         if len(modules) > 1 and single_instance:
             raise SystemExit(f"Cannot have more than one {template_type} per top")
         for module in modules:
-            log.info(f"Generating {template_type} with ipgen")
+            instance_type = module["type"]
+            log.info(f"Generating {instance_type} with ipgen from template {template_type}")
```

Comment:
Nit. If you use single quotes for the f-string, you can use double quotes within the code blocks. Saves a temporary.
```suggestion
            log.info(f'Generating {module["type"]} with ipgen from template {template_type}')
```